### PR TITLE
[Enhancement] Add UT tools to verify the procedures of replay & image

### LINKED ISSUE #8942: Add UT tools to verify the procedures of replay & image
## Enhancement

When we change some metadata in FE, there are two more things we need to do besides just changing the memory.

1. Persist the change to a journal in order to replay to the followers.
2. Persist current metadata to an image when a checkpoint is triggered to make.

Bugs for those two procedures can only be triggered when Frontend is restarted, which makes it very hard to find. Yet developers seldom write unit tests for the two procedures above because of the lack of test utilities. 
We should add some UT tools to make the verification of the two procedures easier.

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #8942 

## Problem Summary(Required) ：

1. Add `PseudoImage` to help to verify the procedure of image on metadata.
2. Add `PseudoJournalReplayer` to help verify the procedure of replay on metadata.
3. Use the above tools on some unittest as demo.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Astralidea — DISMISSED** [review]
LGTM

File: fe/fe-core/src/test/java/com/starrocks/utframe/UtFrameUtils.java [hunk 1]
```diff
@@ -575,4 +588,113 @@ private static void unLock(Map<String, Database> dbs) {
...
+    }
+
+    public static void tearDownForPersisTest() {
+        JournalReplayerHelper.tearDown();
+    }
+
+    /**
+     * image helper is used to test if image is wrote correctly.
+     */
+    public static class ImageHelper {
```
Comment (chaoyli) [inline fe/fe-core/src/test/java/com/starrocks/utframe/UtFrameUtils.java:604]: helper is conflict to the start_be.sh. `PseudoImage` may be a better name.

File: fe/fe-core/src/test/java/com/starrocks/utframe/UtFrameUtils.java [hunk 2]
```diff
@@ -575,4 +588,113 @@ private static void unLock(Map<String, Database> dbs) {
...
+         */
+        public DataInputStream getDataInputStream() throws IOException {
+            return new DataInputStream(new ByteArrayInputStream(buffer.getData(), 0, buffer.getLength()));
+        }
+    }
+
+    /**
+     * journal replayer helper is used to test if replayed correctly
+     */
+    public static class JournalReplayerHelper {
```
Comment (chaoyli) [inline fe/fe-core/src/test/java/com/starrocks/utframe/UtFrameUtils.java:637]: PseudoJournalReplayer

**Review by gengjun-git — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/test/java/com/starrocks/utframe/UtFrameUtils.java (hunk 2 above)
Comment (padmejin) [reply fe/fe-core/src/test/java/com/starrocks/utframe/UtFrameUtils.java]: done

File: fe/fe-core/src/test/java/com/starrocks/utframe/UtFrameUtils.java (hunk 1 above)
Comment (padmejin) [reply fe/fe-core/src/test/java/com/starrocks/utframe/UtFrameUtils.java]: done

**Review by Astralidea — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**